### PR TITLE
Detal out

### LINKED ISSUE #32: [TASK] Detailed TEAL output
--------
Issue Description
--------

Add a detailed output to TEAL which shows yearly cash flows and cash flows for each component to assist in debugging and post processing.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [ ] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by worseliz**
--------
Pull Request Description
--------
##### What issue does this change request address?
Produces detailed cashflow output.

##### What are the significant changes in functionality due to this change request?
More data can be visualized from TEAL.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/HERON/wiki/Code-Standards) for details.
- [ ] 4. Automated Tests should pass.
- [x] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large tes.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added, the the analytic documentation must be updated/added.
- [x] 9. If any test used as a basis for documentation examples have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by worseliz** [comment]
I realized I did this on my user manual branch. I tried deleting the files but they're still showing up here even though they are not in my local branch.

File: src/CashFlow_ExtMod.py [hunk 1]
```diff
@@ -90,10 +92,38 @@ def run(self, container, Inputs):
       @ Out, None
     """
     globalSettings = container._globalSettings
+    #print("\n\n\n\n\n",globalSettings)
     components = container._components
     metrics = main.run(globalSettings, components, Inputs)
-    for k, v in metrics.items():
-      setattr(container, k, v)
+    print(metrics.items())
```
Comment (PaulTalbot-INL) [inline src/CashFlow_ExtMod.py]: may want to take this print out

File: src/CashFlows.py [hunk 2]
```diff
@@ -109,6 +113,7 @@ def __init__(self, verbosity=100, **kwargs):
     self._activeComponents = None
     self._metricTarget = None
     self._components = []
+    self._outputtype = None
```
Comment (PaulTalbot-INL) [inline src/CashFlows.py]: this should probably be camel back, as `self._outputType`. Appears in several places.

File: src/main.py [hunk 3]
```diff
@@ -245,6 +245,7 @@ def componentLifeCashflow(comp, cf, variables, lifetimeCashflows, v=100):
         vprint(v, 1, m, '    {y:^{yx}d}, -- N/A -- , -- N/A -- , {c: 1.9e}'.format(y=y,
                                                            yx=yx,
                                                            c=cash))
+  print(lifeCashflow)
```
Comment (PaulTalbot-INL) [inline src/main.py]: this print can be removed

File: src/main.py [hunk 4]
```diff
@@ -580,12 +583,51 @@ def run(settings, components, variables):
   projectLength = getProjectLength(settings, components, v=v)
   vprint(v, 0, m, ' ... project length: {} years'.format(projectLength))
   projectCashflows = projectLifeCashflows(settings, components, lifetimeCashflows, projectLength, v=v)
+  compCashflows = {}
+  for comp in components:
+    tax = comp.getTax() if comp.getTax() is not None else settings.getTax()
+    inflation = comp.getInflation() if comp.getInflation() is not None else settings.getInflation()
+    compProjCashflows = projectComponentCashflows(comp, tax, inflation, lifetimeCashflows[comp.name], projectLength, v=v)
+    compCashflows[comp.name] = compProjCashflows
+  #print("\n\n\n",compCashflows.keys())
```
Comment (PaulTalbot-INL) [inline src/main.py]: print comments in this file should be removed

File: tests/CashFlow_test_Output.xml [hunk 5]
```diff
@@ -0,0 +1,97 @@
+<Simulation verbosity="debug">
+  <TestInfo>
+    <name>CashFlow_test_IRR</name>
```
Comment (PaulTalbot-INL) [inline tests/CashFlow_test_Output.xml]: if this is a new test, update the TestInfo and remove the revisions

**Review by PaulTalbot-INL — CHANGES_REQUESTED** [review]
A few notes to look at; I did not carefully review the documentation changes yet.

**Comment by PaulTalbot-INL** [comment]
I think this branch might need "devel" merged into it to resolve conflicts and catch up some doc changes.

File: src/CashFlow_ExtMod.py [hunk 6]
```diff
@@ -90,10 +92,42 @@ def run(self, container, Inputs):
       @ Out, None
     """
     globalSettings = container._globalSettings
+    #print("\n\n\n\n\n",globalSettings)
```
Comment (PaulTalbot-INL) [inline src/CashFlow_ExtMod.py]: this line can be removed, I suspect

File: src/CashFlow_ExtMod.py [hunk 7]
```diff
@@ -90,10 +92,42 @@ def run(self, container, Inputs):
...
+        else:
+          blank = []
+          blank.append(v)
+          for x in range(projectLife-1):
+            blank.append(0)
+          blank = np.array(blank)
+
+          setattr(container, f'{k}', blank)
+    else:
+          for k, v in metrics.items():
```
Comment (PaulTalbot-INL) [inline src/CashFlow_ExtMod.py:118]: there's a few too many spaces in this tab

File: src/CashFlows.py [hunk 8]
```diff
@@ -258,6 +266,14 @@ def getTax(self):
     """
     return self._tax
 
+  def getOutput(self):
+    """
+    Get output Type
```
Comment (PaulTalbot-INL) [inline src/CashFlows.py]: these docstring lines should be tabbed in 2 spaces, 1 indent in from the triple-quotes

File: src/main.py [hunk 9]
```diff
@@ -580,24 +583,54 @@ def run(settings, components, variables):
...
-  if 'NPV' in indicators:
-    metric = NPV(components, projectCashflows, projectLength, settings.getDiscountRate(), v=v)
-    results['NPV'] = metric
-  if 'IRR' in indicators:
-    metric = IRR(components, projectCashflows, projectLength, v=v)
-    results['IRR'] = metric
-  if 'PI' in indicators:
-    metric = PI(components, projectCashflows, projectLength, settings.getDiscountRate(), v=v)
-    results['PI'] = metric
+  OutputType = settings.getOutput()
```
Comment (PaulTalbot-INL) [inline src/main.py]: I think `OutputType` should be `outputType` technically. I don't make these rules.

**Review by PaulTalbot-INL — CHANGES_REQUESTED** [review]
A few cleanup changes

**Review by PaulTalbot-INL — APPROVED** [review]

**Comment by PaulTalbot-INL** [comment]
Approved for merge when tests pass, if no further changes needed.

**Pull request merged**